src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -129,6 +133,19 @@ public void Commit(ILogger log, bool forceWrite)
                 }
             }
 
+            // Always write the tool lock files
+            foreach (var toolRestoreResult in ToolRestoreResults)
+            {
+                if (toolRestoreResult.LockFilePath != null)
+                {
+                    log.LogDebug($"Writing tool lock file to disk. Path: {toolRestoreResult.LockFilePath}");
+
+                    var lockFileDirectory = Path.GetDirectoryName(toolRestoreResult.LockFilePath);
```

Comment:
I decided to start with the simplest approach. If we see a performance problem in redoing the tool restore, we can improve this later. Let me know if you disagree and we can pull it in now.